common/network-common/src/main/java/org/apache/spark/network/util/TransportConf.java

```diff
@@ -363,4 +363,26 @@ public boolean useOldFetchProtocol() {
     return conf.getBoolean("spark.shuffle.useOldFetchProtocol", false);
   }
 
+  /**
+   * The minimum size of a chunk when dividing a merged shuffle file into multiple chunks during
+   * push-based shuffle.
+   * A merged shuffle file consists of multiple small shuffle blocks. Fetching the
+   * complete merged shuffle file in a single response increases the memory requirements for the
+   * clients. Instead of serving the entire merged file, the shuffle service serves the
+   * merged file in `chunks`. A `chunk` constitutes few shuffle blocks in entirety and this
+   * configuration controls how big a chunk can get. A corresponding index file for each merged
+   * shuffle file will be generated indicating chunk boundaries.
+   */
+  public int minChunkSizeInMergedShuffleFile() {
+    return Ints.checkedCast(JavaUtils.byteStringAsBytes(
+      conf.get("spark.shuffle.server.minChunkSizeInMergedShuffleFile", "2m")));
```

Comment:
One thing we discussed internally is whether this config should be a server side config or a client side config.
As @otterc mentioned, there are multiple reasons we break a merged shuffle partition file into multiple smaller chunks.
One of the biggest reasons is to parallelize fetching shuffle data and task execution.
If we have a multi-GB merged shuffle partition and the client is fetching it as a single block, then the client would wait until it fetches its entirety before handing off to the task processing logic to process the block, which is not ideal.
The question is that whether size of the chunk should be a global configuration on the server side, irrespective of individual applications, or a Spark app configuration so users can fine tune it.
We currently make it a server side config so we don't introduce another parameter for users to tune.
Want to get inputs from the community on this as well.
cc @Ngone51 @attilapiros @jiangxb1987 @tgravescs

Reply:
Thanks for making the comment here as I was just wondering the same thing.
How much difference have you seen in tuning this parameter?   I assume if its not a static here for the server, then it would have to be passed in as a parameter during fetching.

Reply:
If it's not static, we would make it as a parameter passed during initial executor registration.
This would perhaps introduce a new registration RPC to maintain backward compatibility with the existing RegisterExecutor RPC, and make this chunk size a global config for all shuffles of that application.

In terms of difference, we don't see much impact on the performance side if this size is set to something reasonable, in a few MB range.
When we set it too large, like above 5 MB, it could have slight memory implications if the other parameters like `maxBytesInFlight`, `maxReqsInFlight`, and `maxBlocksInFlightPerAddress` are not set accordingly.
If an executor runs multiple reducers concurrently, each fetching chunks from a merged shuffle partition, then these reducers potentially need to fetch 100s or even 1000+ MB-sized blocks.
This could be a very different block fetching pattern compared with fetching the original blocks.
An inappropriate setting for parameters `maxBytesInFlight`, `maxReqsInFlight`, and `maxBlocksInFlightPerAddress` could lead to more data being fetched concurrently, thus increasing memory consumptions on the client side.

We mitigated this by reducing the default chunk size to 2mb, and also adjusted our settings for the default values for the other shuffle related parameters.

Reply:
I think it is fine to keep it on the server side first. Especially as based on the aboves it does not seem to have huge impact right now (and of course latter when we will have more experience with push-based shuffle this decision can be revisited).

Reply:
yeah agree, I think leave on server side.

Reply:
+1 to leave on the server side.